### ISSUE TITLE
Support to return a custom route instead of custom view to handle the 2FA Code retry.

### ISSUE THREAD

**Issue by graphics-et-al** [Laragear/TwoFactor#47]
### Please check these requirements

- [X] This feature helps everyone using this package
- [X] It's feasible and maintainable
- [X] It's non breaking
- [ ] I issued a PR with the implementation (optional)

### Description

This would enable support for things like Inertiajs where views aren't really used a such but if we can use a route, it's more flexible. Would be super handy in my current project :)

### Code sample

```php
instead of
 $attempt = Auth2FA::attempt($request->only('email', 'password'), $request->filled('remember'))->view('custom2faview');

have 
 $attempt = Auth2FA::attempt($request->only('email', 'password'), $request->filled('remember'))->route('custom2faroute', [params]);
```

**Comment by DarkGhostHunter** [comment]
Care to make a PR? I used InertaJS long time ago and I'm not sure where it could land throwing a redirect to a route, but it should be something like this on the `TwoFactorLoginHelper.php`:

```php
// ...

} catch (InvalidCodeException $e) {
    $this->flashData($credentials, $remember);

    if ($this->route) {
        redirect()->route($this->route)->withErrors($this->request->has($this->input) ? $e->errors())->throwResponse();
    }

    $this->throwConfirmView($this->input, $this->request->has($this->input) ? $e->errors() : []);
}

```

**Comment by graphics-et-al** [comment]
Sorry: my Git-fu is weak, so don't know how to do a PR :(
I've been able to work around it however.

`/**
     * Attempt to authenticate the request's credentials.
     *
     * @return void
     *
     * @throws \Illuminate\Validation\ValidationException
     */
    public function authenticate()
    {

        $this->ensureIsNotRateLimited();

        try {
            if (Auth::attemptWhen($this->only('email', 'password'), TwoFactor::hasCodeOrFails(), $this['remember'])) {
                // Authentication was successful...
            } else {
                RateLimiter::hit($this->throttleKey());
                //  dd($this->getFlashedData());
                throw ValidationException::withMessages([
                    'email' => trans('auth.failed'),
                ]);
            }
        } catch (InvalidCodeException $e) {
            // did a code come in?
            if ($this->isNotFilled('code')) {
                // flash the credentials
                $this->flashData($this->only('email', 'password'), $this['remember']);
                redirect()->route('frontend.show2faform')->throwResponse();
            } else {
                // go to the 2FA view
                throw ValidationException::withMessages([
                    'code' => $e->getMessage(),
                ]);
            }
        }
        RateLimiter::clear($this->throttleKey());
    }
`

**Comment by DarkGhostHunter** [comment]
Fixed on #73
